src/Sarif.Viewer.VisualStudio.Core/CodeFinding/Internals/CodeFinderBase.cs

```diff
@@ -47,6 +53,19 @@ public CodeFinderBase(string fileContents)
             lineCount = GetLineNumber(endOfFile);
 
             IgnoredSpans = GetIgnoredSpans();
+
+            if (fileContents.Contains("\r\n"))
+            {
+                lineEndings = "\r\n";
+            }
+            else if (fileContents.Contains("\r"))
+            {
+                lineEndings = "\r";
+            }
+            else
+            {
+                lineEndings = "\n";
+            }
```

Comment:
Do files ever contain a mixture of line endings?

Reply:
Not that I've seen, usually files just use 1 type of line ending so that they're easily read by text editors. I could imagine a user could manually add different types of line endings into the same file but I don't think I've seen it before.

https://en.wikipedia.org/wiki/Newline#Representation goes into it deeper but my understanding is that windows uses \r\n, modern *nix based os's like mac and linux use \n, and some tools like older mac os's and some codeql scripts give \r.

Reply:
Makes sense. IMO, if producers are doing something weird like this they're on their own.